packages/icon/src/vaadin-icon.js

```diff
@@ -254,6 +273,49 @@ class Icon extends ThemableMixin(
     }
   }
 
+  async __srcChanged(src) {
+    if (!src) {
+      this.svg = null;
+      return;
+    }
+
+    // Need to add the "icon" attribute to avoid issues as described in
+    // https://github.com/vaadin/web-components/issues/6301
+    this.icon = '';
+
+    if (src.includes('#')) {
+      this.svg = svg`<use href="${src}"/>`;
+    } else {
+      try {
+        const svgData = await this.__fetchSVG(src);
+        if (!Icon.__domParser) {
+          Icon.__domParser = new DOMParser();
+        }
+        const parsedResponse = Icon.__domParser.parseFromString(svgData, 'text/html');
+
+        const svgElement = parsedResponse.querySelector('svg');
+        if (!svgElement) {
+          throw new Error(`SVG element not found on path: ${src}`);
+        }
+
+        this.__viewBox = svgElement.getAttribute('viewBox');
+        this.svg = unsafeSvgLiteral(svgElement.innerHTML);
+      } catch (e) {
+        console.error(e);
+        this.svg = null;
+      }
+    }
+  }
+
+  async __fetchSVG(path) {
+    const data = await this.__fetch(path, { mode: 'cors' });
```

Comment:
Could also add a test to make sure cors mode is used. Note that you can use sinon [stubs](https://sinonjs.org/releases/latest/stubs/) for `icon.__fetch` in the tests: `sinon.stub(icon, '__fetch').resolves({ ok: true, text: () => Promise.resolve(svgSrc) });`. You can then check the call arguments with `icon.__fetch.firstCall.args[1]`